R/cluster.R

```diff
@@ -225,34 +225,40 @@ makeCluster <-
     }
 
     if (grepl("PoolBeingDeleted", response)) {
-      pool <- rAzureBatch::getPool(poolConfig$name)
+      message <- paste(
+        "Cluster '%s' already exists and is being deleted.",
+        "Another cluster with the same name cannot be created",
+        "until it is deleted. Please wait for the cluster to be deleted",
+        "or create one with a different name"
+      )
 
-      cat(sprintf(
-        paste(
-          "Cluster '%s' already exists and is being deleted.",
-          "Another cluster with the same name cannot be created",
-          "until it is deleted. Please wait for the cluster to be deleted",
-          "or create one with a different name"
-        ),
-        poolConfig$name
-      ),
-      fill = TRUE)
+      if (wait == TRUE) {
+        pool <- rAzureBatch::getPool(poolConfig$name)
 
-      while (areShallowEqual(rAzureBatch::getPool(poolConfig$name)$state,
-                             "deleting")) {
-        cat(".")
-        Sys.sleep(10)
-      }
+        cat(sprintf(message,
+                    poolConfig$name),
+            fill = TRUE)
 
-      cat("\n")
+        while (areShallowEqual(rAzureBatch::getPool(poolConfig$name)$state,
```

Comment:
Do we need areShallowEqual here? Does '==' not work?

Reply:
This is existing code that I didn't modify, but I tested that '==' works fine, replaced it with '=='